channel.go

```diff
@@ -0,0 +1,157 @@
+package pusher
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+)
+
+// Channel represents a subscription to a Pusher channel.
+type Channel interface {
+	// Subscribe attempts to subscribe to the channel if the subscription is not
+	// already active. Authentication will be attempted for private and presence
+	// channels. Note that a nil error does not mean that the subscription was
+	// succesful, just that the subscription request was sent.
+	Subscribe() error
+	// Unsubscribe attempts to unsubscribe from the channel. Note that a nil error
+	// does not mean that the unsubscription was succesful, just that the request
+	// was sent.
+	Unsubscribe() error
+	// Bind returns a channel to which all the data from all matching events received
+	// on the channel will be sent.
+	Bind(event string) chan json.RawMessage
+	// Unbind removes bindings for an event. If chans are passed, only those bindings
+	// will be removed. Otherwise, all bindings for an event will be removed.
+	Unbind(event string, chans ...chan json.RawMessage)
+	// SendEvent sends an event to the channel.
+	Trigger(event string, data interface{}) error
```

Comment:
Also, thoughts on adding comments to the Trigger and HandleEvent functions to be consistent with the comments/annotations for all the other function definitions above?